contracts/bonding/BondingManager.sol

```diff
@@ -332,68 +335,135 @@ contract BondingManager is ManagerProxyTarget, IBondingManager {
+        require(_amount > 0);
+        // Amount to unbond must be less than or equal to current bonded amount 
+        require(_amount <= del.bondedAmount);
 
-        // Transition to unbonding phase
-        del.withdrawRound = currentRound.add(unbondingPeriod);
+        uint256 currentRound = roundsManager().currentRound();
+        uint256 withdrawRound = currentRound.add(unbondingPeriod);
+        uint256 unbondingLockId = del.nextUnbondingLockId;
+
+        // Create new unbonding lock
+        del.unbondingLocks[unbondingLockId] = UnbondingLock({
+            amount: _amount,
+            withdrawRound: withdrawRound
+        });
+        // Increment ID for next unbonding lock
+        del.nextUnbondingLockId = unbondingLockId.add(1);
+        // Decrease delegator's bonded amount
+        del.bondedAmount = del.bondedAmount.sub(_amount);
         // Decrease delegate's delegated amount
-        delegators[del.delegateAddress].delegatedAmount = delegators[del.delegateAddress].delegatedAmount.sub(del.bondedAmount);
+        delegators[del.delegateAddress].delegatedAmount = delegators[del.delegateAddress].delegatedAmount.sub(_amount);
         // Update total bonded tokens
-        totalBonded = totalBonded.sub(del.bondedAmount);
-
-        if (transcoderStatus(msg.sender) == TranscoderStatus.Registered) {
-            // If caller is a registered transcoder, resign
-            // In the future, with partial unbonding there would be a check for 0 bonded stake as well
-            resignTranscoder(msg.sender);
+        totalBonded = totalBonded.sub(_amount);
+
+        if (transcoderStatus(del.delegateAddress) == TranscoderStatus.Registered && (del.delegateAddress != msg.sender || del.bondedAmount > 0)) {
+            // A transcoder's delegated stake within the registered pool needs to be decreased if:
+            // - The caller's delegate is a registered transcoder
+            // - Caller is not delegated to self OR caller is delegated to self and has a non-zero bonded amount
+            // If the caller is delegated to self and has a zero bonded amount, it will be removed from the 
+            // transcoder pool so its delegated stake within the pool does not need to be decreased
+            transcoderPool.updateKey(del.delegateAddress, transcoderPool.getKey(del.delegateAddress).sub(_amount), address(0), address(0));
         }
 
-        if (del.delegateAddress != msg.sender && transcoderStatus(del.delegateAddress) == TranscoderStatus.Registered) {
-            // If delegate is not self and is a registered transcoder, decrease its delegated stake
-            // We do not need to decrease delegated stake if delegate is self because we would have already removed self
-            // from the transcoder pool
-            transcoderPool.updateKey(del.delegateAddress, transcoderPool.getKey(del.delegateAddress).sub(del.bondedAmount), address(0), address(0));
+        // Check if delegator is still bonded
```

Comment:
This comment is slightly confusing to me. We're actually checking if they should no longer be bonded. Maybe:

`// Check if delegator has zero bondedAmount remaining, and if so reset their delegation status`